### PR TITLE
Home Page

### LINKED ISSUE #60: Home Page
The homepage needs a general layout.

Feel free to make any suggestions, or mockups here.

### PR DESCRIPTION
**Pull request by Flizzyclone**
Adds a proper design for the home page, looks like below:
Desktop (1920x1080):
image

Mobile (Pixel 5):
image

Tablet (Ipad Air)
image

"More" buttons (defined by a "modal" boolean prop) lead to a modal where you can select all maps:
image

Summary of Code Changes:
- Redid page.svelte for the home page completely.
- Added Login icon
- Added "MoreMapsModal" for seeing all maps for a country. The modal is constructed by reading data from MapList.json which contains information about maps organized by country "key". For usa, its 'usa'.
- Created HomePageSidebar, contains the social links & updates.
- Created MapSelectionCard, constructed by props passed to it, MapSelectionTitle (similar to ModalTitle), and mapSelectionGrid (simple div)
- Data for constructing the map selection section of the home page using MapSelectionTitle, MapSelectionCard, MapSelectionGrid is grabbed from MapSelection.json. At some point, we could switch this to an API call if we like.
- Added countrybackgrounds in static so page.svelte can access them.

Login button and updating the "updates" section via API are not yet functional.
Closes #60

### CODE REVIEW

**Comment by Fish1** [comment]
Looking good! A couple things I just noticed skimming over it. But we can probably just offload these into different issues.

1) The social links have wrapping enabled, causing this.
image

2) I think the medium size, like on your iPad is a bit strange. The Update sidebar is larger than the main content of the page. Maybe we should make the update sidebar be horizontal on the bottom of the screen.
image

I'll do a review of the code in a bit.

**Comment by Flizzyclone** [comment]
> Looking good! A couple things I just noticed skimming over it. But we can probably just offload these into different issues.
> 
> 1. The social links have wrapping enabled, causing this.
>    image
> 2. I think the medium size, like on your iPad is a bit strange. The Update sidebar is larger than the main content of the page. Maybe we should make the update sidebar be horizontal on the bottom of the screen.
>    image
> 
> I'll do a review of the code in a bit.

Good idea with having the updates be on the bottom, I was struggling with what to do with that size.

File: apps/yapms/src/lib/assets/other/MapList.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[
+	{
+		"key": "usa",
+		"label": "United States",
+		"links": [
```
Comment (Fish1) [inline apps/yapms/src/lib/assets/other/MapList.json]: This file should be put into a more descriptive folder. Not just a folder called "other".

File: apps/yapms/src/lib/assets/other/MapSelection.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
+[
+	{
+		"title": "Popular Maps",
+		"cards": [
```
Comment (Fish1) [inline apps/yapms/src/lib/assets/other/MapSelection.json]: This file should be put into a more descriptive folder. Not just a folder called "other".

File: apps/yapms/src/lib/components/homePage/MapSelectionCard.svelte [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+			open: true
+		});
+    }
+</script>
+
+<div class="card card-bordered w-80 md:w-92 h-48 lg:h-52 bg-base-100 shadow-xl image-full">
+    <figure><img src={bg} alt={alt} /></figure>
+    <div class="card-body items-center text-center">
+        <h2 class="card-title text-primary-content">{name}</h2>
+        <div class="grid gap-4 {doubleCols ? "grid-cols-2": ""}">
```
Comment (Fish1) [inline apps/yapms/src/lib/components/homePage/MapSelectionCard.svelte]: Dynamic class should use sveltes dynamic class capabilities.

you can do ```class:grid-cols-2={doubleCols}```

File: apps/yapms/src/routes/+page.svelte [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
-				<a class="link" href="/app/usa/presidential/2022">House</a>
-				<a class="link" href="/app/usa/presidential/2022">Governor</a>
-			</div>
-			<div class="card-actions justify-end">
-				<button class="btn btn-sm btn-primary">Buy Now</button>
-			</div>
-		</div>
+<svelte:head>
+	<title>YAPms - Yet Another Political Map Simulator</title>
+</svelte:head>
```
Comment (Fish1) [inline apps/yapms/src/routes/+page.svelte:14]: The svelte:head tag should go directly below the <script> tag. Just for organization.

File: apps/yapms/src/routes/+page.svelte [hunk 5]
```diff
@@ -1,20 +1,42 @@
...
+	import HomePageSidebar from "$lib/components/homePage/homePageSidebar.svelte";
+	import MapSelectionGrid from "$lib/components/homePage/mapSelectionGrid.svelte";
+	import MapSelectionTitle from "$lib/components/homePage/mapSelectionTitle.svelte";
+	import Login from '$lib/icons/Login.svelte';
+	import mapSelectSections from '$lib/assets/other/MapSelection.json';
+	import MapSelectionCard from "$lib/components/homePage/MapSelectionCard.svelte";
+	import MoreMapsModal from "$lib/components/modals/moremapsmodal/MoreMapsModal.svelte";
+
+</script>
+<div class="navbar bg-base-200 h-[10%]">
```
Comment (Fish1) [inline apps/yapms/src/routes/+page.svelte]: This ```h-[10%]``` can be removed. I think.

File: apps/yapms/src/routes/+page.svelte [hunk 6]
```diff
@@ -1,20 +1,42 @@
...
-			</div>
-			<div class="card-actions justify-end">
-				<button class="btn btn-sm btn-primary">Buy Now</button>
-			</div>
-		</div>
+<svelte:head>
+	<title>YAPms - Yet Another Political Map Simulator</title>
+</svelte:head>
+
+<div class="flex flex-row w-full h-[90%]">
```
Comment (Fish1) [inline apps/yapms/src/routes/+page.svelte]: This h-[90%] can be removed, also.

**Review by Fish1 — CHANGES_REQUESTED** [review]
Looks good! Just a few organizational and formatting things.

1) All folders should be all lower case, and all components should be Pascal Case.
image

2) The images of the the country backgrounds should be in the src code assets, and imported directly into the code.

image

**Comment by Fish1** [comment]
Oh, sorry, I see why those height percentages were there.

But those should be reformatted into using flex boxes, similar to how it's managed on the app route. I don't think this is the exact solution, but something like it.
image

**Comment by Flizzyclone** [comment]
I used percentages because flex has an odd issue. When scrolling to the bottom of the div with all the selection cards in it, the last one will be cut off.
image

Using flex, I can fix this by having a fake, invisible card below
image
image

I will commit this fix along with the others and can roll it back if you don't think it's desirable.

**Comment by Flizzyclone** [comment]
Not sure why the lint failed on this... pnpm run format produces no issues and it looks like there might have been an issue copying files on Github's end.

**Comment by Flizzyclone** [comment]
@Fish1, countrybackgrounds must be in static. Otherwise, components within the HTML cannot access it when building. Doesn't work like an SVG where we return the svg data itself from the component unfortunately.
`yapms:build: Error: Not found: /src/lib/assets/countrybackgrounds/usa.webp
yapms:build:     at resolve (file:///home/runner/work/yapms/yapms/apps/yapms/.svelte-kit/output/server/index.js:3017:18)
yapms:build:     at resolve (file:///home/runner/work/yapms/yapms/apps/yapms/.svelte-kit/output/server/index.js:2904:34)
yapms:build:     at options.hooks.handle (file:///home/runner/work/yapms/yapms/apps/yapms/.svelte-kit/output/server/index.js:3134:59)
yapms:build:     at respond (file:///home/runner/work/yapms/yapms/apps/yapms/.svelte-kit/output/server/index.js:2902:42)
yapms:build: file:///home/runner/work/yapms/yapms/node_modules/.pnpm/@sveltejs+kit@1.0.1_svelte@3.55.0+vite@4.0.3/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:36
yapms:build: 				throw new Error(format(details));
yapms:build: 				      ^
yapms:build: 
yapms:build: Error: 404 /src/lib/assets/countrybackgrounds/usa.webp (linked from /)
yapms:build:     at file:///home/runner/work/yapms/yapms/node_modules/.pnpm/@sveltejs+kit@1.0.1_svelte@3.55.0+vite@4.0.3/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:36:11
yapms:build:     at save (file:///home/runner/work/yapms/yapms/node_modules/.pnpm/@sveltejs+kit@1.0.1_svelte@3.55.0+vite@4.0.3/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:349:4)
yapms:build:     at visit (file:///home/runner/work/yapms/yapms/node_modules/.pnpm/@sveltejs+kit@1.0.1_svelte@3.55.0+vite@4.0.3/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:199:3)`

Error building when countrybackgrounds in src

**Comment by Fish1** [comment]
This is what I tried on the home page, it should let you remove that blank card at the bottom.

```html
<div class="flex flex-col h-full overflow-hidden">

	<div class="navbar bg-base-200">
		<div class="navbar-start"></div>
		<div class="navbar-center">
			<h1 class="text-2xl font-bold m-auto hidden lg:inline">Yet Another Political Map Simulator</h1>
			<h1 class="text-2xl font-bold m-auto inline lg:hidden">YAPms</h1>
		</div>
		<div class="navbar-end">
			<button class="btn px-8 btn-primary mr-2 hidden md:inline">Login</button>
			<button class="btn btn-square mr-2 inline md:hidden"><Login class="h-8 m-auto"/></button>
		</div>
	</div>

	<div class="flex flex-row h-full overflow-hidden">
		<HomePageSidebar/>
		<div class="divider md:divider-horizontal ml-0 w-0" />
		<div class="flex-1 md:pr-5 overflow-scroll">
			{#each mapSelectSections as section}
				<MapSelectionTitle title={section.title}></MapSelectionTitle>
				<MapSelectionGrid>
					{#each section.cards as card}
						<MapSelectionCard name={card.name} background={card.bg} alt={card.alt} doubleCols={card.doubleCols} links={card.links}></MapSelectionCard>
					{/each}
				</MapSelectionGrid>
			{/each}
		</div>
	</div>
	
</div>```

**Comment by Fish1** [comment]
Running the image imports through rollup imports would be ideal though, as it provides cache breaking features, and you can run them through optimizes.

But hard coding the URL should be fine for now.

**Review by Fish1 — CHANGES_REQUESTED** [review]
See above two posts.

**Comment by Flizzyclone** [comment]
Unsure of what magic you did with the first post but it seems to have worked, and I have implemented it.

**Review by Fish1 — APPROVED** [review]
LGTM

**Pull request merged**